src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx

```diff
@@ -0,0 +1,132 @@
+import { Box, Button, Join, Spacer, Text } from "@artsy/palette"
+import { FC, useReducer } from "react"
+import { Artists } from "./Components/Artists"
+import { Interests } from "./Components/Interests"
+import { Goals } from "./Components/Goals"
+
+export type State = {
+  goals: string[]
+  goalsFreeText: string
+  interests: string[]
+  interestsFreeText: string
+  artists: string[]
+  artistsFreeText: string
+}
+
+const initialState: State = {
+  goals: [],
+  goalsFreeText: "",
+  interests: [],
+  interestsFreeText: "",
+  artists: [],
+  artistsFreeText: "",
+}
+
+export type Action =
+  | { type: "RESET" }
+  // goals
+  | { type: "TOGGLE_GOAL"; goal: string }
+  | { type: "UNSURE_GOALS" }
+  | { type: "SET_FREETEXT_GOALS"; text: string }
+  // interests
+  | { type: "TOGGLE_INTEREST"; interest: string }
+  | { type: "UNSURE_INTERESTS" }
+  | { type: "SET_FREETEXT_INTEREST"; text: string }
+  // artists
+  | { type: "TOGGLE_ARTIST"; artist: string }
+  | { type: "UNSURE_ARTISTS" }
+  | { type: "SET_FREETEXT_ARTISTS"; text: string }
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "RESET":
+      return initialState
+
+    case "TOGGLE_GOAL":
+      const goals = state.goals.includes(action.goal)
+        ? state.goals.filter(goal => goal !== action.goal)
+        : [...state.goals, action.goal]
+      return { ...state, goals }
+    case "UNSURE_GOALS":
+      return { ...state, goals: [] }
+    case "SET_FREETEXT_GOALS":
+      return { ...state, goalsFreeText: action.text }
```

Comment:
A lot of repetition across the three questions here, but the same idea for each: using state reducers to toggle or clear the suggestion pills; or to set the freetext field.